bigbluebutton-html5/imports/ui/components/presentation/component.jsx

```diff
@@ -1,78 +1,246 @@
+import WhiteboardToolbarContainer from '/imports/ui/components/whiteboard/whiteboard-toolbar/container';
 import PollingContainer from '/imports/ui/components/polling/container';
+import CursorWrapperContainer from './cursor/cursor-wrapper-container/container';
 import AnnotationGroupContainer from '../whiteboard/annotation-group/container';
-import Cursor from './cursor/component';
 import PresentationToolbarContainer from './presentation-toolbar/container';
+import PresentationOverlayContainer from './presentation-overlay/container';
 import Slide from './slide/component';
 import styles from './styles.scss';
 
+
 export default class PresentationArea extends React.Component {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super();
+
+    this.state = {
+      paperWidth: 0,
+      paperHeight: 0,
+      showSlide: false,
+    };
+
+    this.getSvgRef = this.getSvgRef.bind(this);
   }
 
+  componentDidMount() {
+    // adding an event listener to scale the whiteboard on 'resize' events sent by chat/userlist etc
+    window.addEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+
+    this.getInitialPaperSizes();
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', () => {
+      setTimeout(this.handleResize.bind(this), 0);
+    });
+  }
+
+  // returns a ref to the svg element, which is required by a WhiteboardOverlay
+  // to transform screen coordinates to svg coordinate system
+  getSvgRef() {
+    return this.svggroup;
+  }
+
+  getPaperSizes() {
+    const { presentationPaper, whiteboardSizeAvailable } = this;
+    const paperSizes = {};
+
+    if (presentationPaper) {
+      // if a user is a presenter - this means there is a whiteboardToolBar on the right
+      // and we have to get the width/height of the whiteboardSizeAvailable
+      // (inner hidden div with absolute position)
+      let clientHeight;
+      let clientWidth;
+      if (this.props.userIsPresenter) {
+        clientHeight = whiteboardSizeAvailable.clientHeight;
+        clientWidth = whiteboardSizeAvailable.clientWidth;
+      // user is not a presenter - we can get the sizes of the presentationPaper
+      // direct parent of the svg wrapper
+      } else {
+        clientHeight = presentationPaper.clientHeight;
+        clientWidth = presentationPaper.clientWidth;
+      }
+
+      paperSizes.paperHeight = clientHeight;
+      paperSizes.paperWidth = clientWidth;
+    }
+    return paperSizes;
+  }
+
+  getInitialPaperSizes() {
+    // determining the paperWidth and paperHeight (available space for the svg) on the initial load
+    const paperSizes = this.getPaperSizes();
+    if (Object.keys(paperSizes).length > 0) {
+      // setting the state of the paperWidth and paperHeight (available space for the svg)
+      // and set the showSlide to true to start rendering the slide
+      paperSizes.showSlide = true;
+      this.setState(paperSizes);
```

Comment:
It doesn't merge, but it may batch the setState calls, and the class will be updated just once, that's true.
It would work fine in this particular case, but I'm trying to avoid calling `setState` more than once withing one cycle. It's just a good practice which helps you to avoid unexpected bugs, which you can get if you change the same property more than once within one cycle.
Maybe I just need to change the variable name to `stateObject`, so that it was more clear that it has not only sizes of the whiteboard area.